presto-spi/src/main/java/io/prestosql/spi/block/Int128ArrayBlockEncoding.java

```diff
@@ -54,13 +67,33 @@ public Block readBlock(BlockEncodingSerde blockEncodingSerde, SliceInput sliceIn
         boolean[] valueIsNull = decodeNullBits(sliceInput, positionCount).orElse(null);
 
         long[] values = new long[positionCount * 2];
-        for (int position = 0; position < positionCount; position++) {
-            if (valueIsNull == null || !valueIsNull[position]) {
-                values[position * 2] = sliceInput.readLong();
-                values[(position * 2) + 1] = sliceInput.readLong();
+        if (valueIsNull == null) {
+            sliceInput.readBytes(Slices.wrappedLongArray(values));
+        }
+        else {
+            int nonNullPositionCount = sliceInput.readInt();
+            sliceInput.readBytes(Slices.wrappedLongArray(values, 0, nonNullPositionCount * 2));
+            int position = 2 * (nonNullPositionCount - 1);
+            for (int i = positionCount - 1; i >= 0 && position >= 0; i--) {
```

Comment:
Why not use similar 8-bits null check as in in other serdes?

Reply:
It turns out to be slower. I guess more data means less time spent on null checking and more on actual memory copying.